api/v1/qdrantcluster_types.go

```diff
@@ -181,9 +181,40 @@ func (s QdrantClusterSpec) GetServicePerNode() bool {
 }
 
 type GPU struct {
-	// GPUType specifies the type of the GPU to use.
+	// GPUType specifies the type of the GPU to use. If set, GPU indexing is enabled.
 	// +kubebuilder:validation:Enum=nvidia;amd
 	GPUType GPUType `json:"gpuType"`
+	// ForceHalfPrecision for `f32` values while indexing.
+	// `f16` conversion will take place
+	// only inside GPU memory and won't affect storage type.
+	// +kubebuilder:default=false
+	ForceHalfPrecision bool `json:"forceHalfPrecision"`
+	// DeviceFilter for GPU devices by hardware name. Case-insensitive.
+	// List of substrings to match against the gpu device name.
+	// Example: [- "nvidia"]
+	// If not specified, all devices are accepted.
+	// +optional
+	DeviceFilter []string `json:"deviceFilter,omitempty"`
+	// Devices is a List of explicit GPU devices to use.
+	// If host has multiple GPUs, this option allows to select specific devices
+	// by their index in the list of found devices.
+	// If `deviceFilter` is set, indexes are applied after filtering.
+	// If not specified, all devices are accepted.
+	// +optional
+	Devices []string `json:"devices,omitempty"`
+	// ParallelIndexes is the number of parallel indexes to run on the GPU.
+	// +kubebuilder:default=1
+	ParallelIndexes int `json:"parallelIndexes"`
+	// GroupsCount is the amount of used vulkan "groups" of GPU.
+	// In other words, how many parallel points can be indexed by GPU.
+	// Optimal value might depend on the GPU model.
+	// Proportional, but doesn't necessary equal to the physical number of warps.
+	// Do not change this value unless you know what you are doing.
+	// +optional
+	GroupsCount int `json:"groupsCount,omitempty"`
```

Comment:
Qdrant docs says that the default should be 512?

Also, see above, we can forbid negative values.

Reply:
Yes, forbidding negative values makes sense

In this case, I don't want to set a default value here in case core decides to change the qdrant default.